src/plugins/echo_plugin/echo_plugin.cc

```diff
@@ -222,21 +204,23 @@ void EchoPlugin::RegisterEchoChannel() {
 
   for (const auto& topic_meta_itr : topic_meta_list) {
     const auto& topic_meta = topic_meta_itr.second;
-    EchoFunc echo_func;
-    if (options_.executor.empty()) {
-      echo_func = [this, echo_type{topic_meta.echo_type}](
-                      MsgWrapper& msg_wrapper) {
-        Echo(msg_wrapper, echo_type);
-      };
-    } else {
-      echo_func = [this, echo_type{topic_meta.echo_type}](
-                      MsgWrapper& msg_wrapper) {
-        executor_.Execute([this, msg_wrapper{std::move(msg_wrapper)}, echo_type]() mutable {
-          Echo(msg_wrapper, echo_type);
-        });
-      };
-    }
-
+    EchoFunc echo_func = [this, echo_type{topic_meta.echo_type}](MsgWrapper& msg_wrapper) {
```

Comment:
do not use echo_func as Intermediate steps. Set the `sub_wrapper.callback` directly